src/App.tsx

```diff
@@ -1,18 +1,44 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination/Pagination';
+import { ItemsList } from './components/ItemsList';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+enum DefaultPageValues {
+  StartPage = 1,
+  DefaultPageSize = 5,
+  TotalPages = 42,
+}
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(DefaultPageValues.StartPage);
+  const [itemsPerPage, setPerPage] = useState(
+    DefaultPageValues.DefaultPageSize,
+  );
+  const startIndex = currentPage * itemsPerPage - itemsPerPage + 1;
+  const endIndex = currentPage * itemsPerPage > DefaultPageValues.TotalPages
+    ? DefaultPageValues.TotalPages
+    : currentPage * itemsPerPage;
+
+  const handleItemsPerPage = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(Number(event?.currentTarget.value));
+    setCurrentPage(DefaultPageValues.StartPage);
+  };
+
+  const paginationProps = () => (
+    {
+      total: DefaultPageValues.TotalPages,
+      perPage: itemsPerPage,
+      currentPage,
+      onPageChange: (pageNumber: number) => setCurrentPage(pageNumber),
+    }
+  );
```

Comment:
In this cimment https://github.com/mate-academy/react_pagination/pull/599#discussion_r1318262766 t, Dmytro had in mind to create a function for the callback, not for all props :)